pkgs/development/python-modules/woob/default.nix

```
@@ -0,0 +1,87 @@
+{ lib, buildPythonPackage, fetchPypi, isPy27
+, Babel
+, colorama
+, cssselect
+, dateutil
+, feedparser
+, futures
+, gdata
+, gnupg
+, google-api-python-client
+, html2text
+, libyaml
+, lxml
+, mechanize
+, nose
+, pdfminer
+, pillow
+, prettytable
+, pyqt5
+, pyyaml
+, requests
+, simplejson
+, termcolor
+, unidecode
+}:
+
+buildPythonPackage rec {
+  pname = "woob";
+  version = "3.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "09hpxy5zhn2b8li0xjf3zd7s46lawb0315p5mdcsci3bj3s4v1j7";
+  };
+
+  postPatch = ''
+    # Disable doctests that require networking:
+    sed -i -n -e '/^ *def \+pagination *(.*: *$/ {
+      p; n; p; /"""\|'\'\'\'''/!b
+
+      :loop
+      n; /^ *\(>>>\|\.\.\.\)/ { h; bloop }
+      x; /^ *\(>>>\|\.\.\.\)/bloop; x
+      p; /"""\|'\'\'\'''/b
+      bloop
+    }; p' woob/browser/browsers.py woob/browser/pages.py
+  '';
+
+  checkInputs = [ nose ];
+
+  nativeBuildInputs = [ pyqt5 ];
+
+  propagatedBuildInputs = [
+    Babel
+    colorama
+    cssselect
+    dateutil
+    feedparser
+    gdata
+    gnupg
+    google-api-python-client
+    html2text
+    libyaml
+    lxml
+    mechanize
+    pdfminer
+    pillow
+    prettytable
+    pyqt5
+    pyyaml
+    requests
+    simplejson
+    termcolor
+    unidecode
+  ] ++ lib.optionals isPy27 [ futures ];
+
+  checkPhase = ''
+    nosetests
+  '';
+
+  meta = {
+    homepage = "https://woob.tech";
+    description = "Collection of applications and APIs to interact with websites without requiring the user to open a browser";
+    license = lib.licenses.agpl3;
+    maintainers = [ lib.maintainers.DamienCassou ];
```

```suggestion
    license = licenses.agpl3;
    maintainers = [ maintainers.DamienCassou ];
```

agpl3Only or agpl3Plus?

---

they switched to *L*GPL3 2 years ago: https://gitlab.com/woob/woob/-/blob/master/COPYING

---

According to https://gitlab.com/woob/woob/-/blob/master/woob/exceptions.py#L10 it is agpl3Plus.

---

you are right. I also confirmed with the authors on IRC.